### DISCUSSION TITLE
Filtering geometry that is too small to be visible at the current zoom level

### DISCUSSION THREAD

**Discussion by sharkAndshark** [maplibre/martin#731] [Ideas]
Hi. It's just some random ideas.
For automatically discovered data sources, Martin will fetch a large amount of data when the zoom level is low, such as 0 to 4. At these zoom levels, the tiles are large and contain an overload of information on the map.Is it possible to remove polygons or linestrings whose area / length is not bigger than a single pixel in size at the current zoom level? 

| scale | Marin table source | Martin function source with filtering |
| ---------------------- | ---------------------  | ---------------------- |
|1:250000 | ![image](https://github.com/maplibre/martin/assets/42399654/09bd880c-b761-419b-9172-8b50ace7e925)| ![image](https://github.com/maplibre/martin/assets/42399654/061baabb-334f-4f22-93e5-25743678917b)|
|1:100000|![image](https://github.com/maplibre/martin/assets/42399654/e78370e0-52d3-48bd-a7ad-14fa497dc4e3)|![image](https://github.com/maplibre/martin/assets/42399654/b52af32c-fbb4-4406-811c-cd461c0bc543)|

The function source
```sql
CREATE 
	OR REPLACE FUNCTION test_filtering ( z INTEGER, x INTEGER, y INTEGER ) RETURNS bytea AS $$ DECLARE
	mvt bytea;
BEGIN
	SELECT INTO
		mvt ST_AsMVT ( tile, 'function_zxy_query', 4096, 'geom' ) 
	FROM
		(
		SELECT
			ST_AsMVTGeom ( ST_Transform ( ST_CurveToLine ( geom ), 3857 ), ST_TileEnvelope ( z, x, y ), 4096, 64, TRUE ) AS geom 
		FROM
			"SHANGHAI_BUILDINGS" 
		WHERE
			geom && ST_Transform ( ST_TileEnvelope ( z, x, y ), 4326 ) 
			AND ( st_area ( st_transform ( geom, 3857 ) ) >= ( ( 40075016.68 / ( 2 ^ z * 512 ) ) ^ 2 ) ) 
-- 			when the tile size is 512 * 512 pixels, the actual area covered by one pixel would be: (40075016.68 / (2^zoom * 512))^2
		) AS tile 
	WHERE
		geom IS NOT NULL;
	RETURN mvt;

END $$ LANGUAGE plpgsql IMMUTABLE STRICT PARALLEL SAFE;
```

**Comment by nyurik** [comment]
@sharkAndshark I think this is a worthy idea to be implemented, thanks!  I will create an issue from this, but not certain when i can get to it. Want to try to implement it?

**Reply by sharkAndshark** [reply]
Yeah. Ok
